node_settled_handler.go

```diff
@@ -115,51 +110,61 @@ func (p *NodeSettledHandler) subscribeEvents(ctx context.Context) error {
 		if !ok {
 			continue
 		}
-		if !finalEvent.FinalHtlcEvent.Settled {
-			// TODO: Handle final fail?
-
-			continue
-		}
 
 		key := types.CircuitKey{
 			ChanID: event.IncomingChannelId,
 			HtlcID: event.IncomingHtlcId,
 		}
 
-		if err := p.handleFinalHtlc(ctx, key); err != nil {
+		err = p.handleFinalHtlc(
+			ctx, key, finalEvent.FinalHtlcEvent.Settled,
+		)
+		if err != nil {
 			return err
 		}
 	}
 }
 
 func (p *NodeSettledHandler) handleFinalHtlc(ctx context.Context,
-	key types.CircuitKey) error {
+	key types.CircuitKey, settled bool) error {
+
+	var status types.HtlcStatus
```

Comment:
It's only for logging indeed.

If it's annoying, I can either put the logs directly in persistence (maybe not as consistent), or remove some info from the logs (the status)

Reply:
I think you can just log the settled boolean which you already have instead of mapping to status just for logging/err?